util/oidc/oidc.go

```diff
@@ -187,7 +188,14 @@ func NewClientApp(settings *settings.ArgoCDSettings, dexServerAddr string, dexTL
 		encryptionKey:            encryptionKey,
 		clientCache:              cacheClient,
 		azure:                    azureApp{mtx: &sync.RWMutex{}},
-		refreshTokenThreshold:    settings.OIDCRefreshTokenThreshold,
+		domainHint: func() string {
+			oidcConfig := settings.OIDCConfig()
+			if oidcConfig != nil {
+				return oidcConfig.DomainHint
+			}
+			return ""
+		}(),
```

Comment:
I'm not trying to be pricky here but I'd not prefer having an inline function here since it's not testable easily. I'd prefer you create a separate helper function and add unit tests for it. You can get the `domainHint` value before initializing the variable which is a neat and clean approach.